File: src/electionguard_gui/web/services/authorization-service.js
export default {
  _userId: undefined,
  _isAdmin: undefined,
  async getUserId() {
    if (!this._userId) {
      this._userId = await eel.get_user_id()();
    }
    return this._userId;
  },
  async setUserId(id) {
    await eel.set_user_id(id)();
    this._userId = id;
  },
  async isAdmin() {
    if (!this._isAdmin) {
      this._isAdmin = await eel.is_admin()();
    }
    return this._isAdmin;
  },
};

File: src/electionguard_gui/services/authorization_service.py
from typing import Optional
import eel

from electionguard_gui.services.configuration_service import ConfigurationService
from electionguard_gui.services.service_base import ServiceBase


class AuthorizationService(ServiceBase):
    """Responsible for functionality related to authorization and user identify"""

    _is_admin: bool

    def __init__(self, config_service: ConfigurationService) -> None:
        self._is_admin = config_service.get_is_admin()

    # todo: replace state based storage with configparser https://docs.python.org/3/library/configparser.html
    user_id: Optional[str] = None

    def expose(self) -> None:
        eel.expose(self.get_user_id)
        eel.expose(self.set_user_id)
        eel.expose(self.is_admin)

    def get_required_user_id(self) -> str:
        if self.user_id is None:
            raise Exception("User must be logged in")
        return self.user_id

    def get_user_id(self) -> Optional[str]:
        return self.user_id

    def set_user_id(self, user_id: str) -> None:
        self.user_id = user_id

    def is_admin(self) -> bool:
        return self._is_admin
